google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1342,8 +1344,14 @@ def extract_job extract_url, format: nil, compression: nil,
           ensure_service!
           options = { format: format, compression: compression,
                       delimiter: delimiter, header: header, dryrun: dryrun,
-                      job_id: job_id, prefix: prefix, labels: labels }
-          gapi = service.extract_table table_ref, extract_url, options
+                      labels: labels }
+          updater = ExtractJob::Updater.from_options table_ref, extract_url,
+                                                     options
+
+          yield updater if block_given?
+
+          job_gapi = updater.to_gapi
+          gapi = service.extract_table job_id, prefix, job_gapi
```

Comment:
Instead of passing `job_id` and `prefix` to the service call, shouldn't they already be used and present on the updater object? The updater object seems like a fine place to set a custom `job_id` value.

Reply:
Yeah, thinking ahead to the `location` field, this makes sense. I'm missing `job_id` in QueryJob::Updater and LoadJob::Updater, too, though. Maybe let's follow-up with another PR to move this over?

Reply:
If I follow up with the location changes I can make these changes then as well.